[3.x]Upgrade Slurm to 22.05.3

hanwen-pcluste opened the pull request:
In addition to the upgrade:
1. Remove TaskAffinity=no in cgroups.conf. This parameter had been ignored from Slurm 21. In addition, Slurm 22 fails if the parameter still exists.
2. Remove prolog/epilog used as a workaround for cluster without Internet connection.

Signed-off-by: Hanwen <hanwenli@amazon.com>

### Tests
Integration tests in `configs/pcluster3.yaml` and `test_cluster_in_no_internet_subnet` have been passed

### References
This PR should be merged together with https://github.com/aws/aws-parallelcluster/pull/4347

### Checklist
- [x] Make sure you are pointing to **the right branch** and add a label in the PR title (i.e. **2.x** vs **3.x**)
- [x] Check all commits' messages are clear, describing what and why vs how.
- [x] Make sure **to have added unit tests or integration tests** to cover the new/modified code.
- [ ] Check if documentation is impacted by this change.

Please review the [guidelines for contributing](../CONTRIBUTING.md) and [Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

demartinofra commented on cookbooks/aws-parallelcluster-slurm/recipes/config_head_node.rb:
```diff
@@ -192,34 +192,6 @@
-    mode '0755'
-  end
-
-  cookbook_file "#{node['cluster']['slurm']['install_dir']}/etc/pcluster/prolog.d/01-pcluster-prolog" do
```
you might also want to delete the content of this file

hanwen-pcluste replied:
Done

demartinofra commented on cookbooks/aws-parallelcluster-slurm/recipes/config_head_node.rb:
```diff
@@ -192,34 +192,6 @@
-    mode '0755'
-  end
-
-  cookbook_file "#{node['cluster']['slurm']['install_dir']}/etc/pcluster/epilog.d/01-pcluster-epilog" do
```
same here

hanwen-pcluste replied:
Done

jdeamicis commented on cookbooks/aws-parallelcluster-slurm/templates/default/slurm/slurm.conf.erb:
```diff
@@ -40,19 +40,6 @@ SuspendTimeout=120
-Prolog=<%= node['cluster']['slurm']['install_dir'] %>/etc/pcluster/prolog.d/*
-PrologFlags=alloc,serial
-SchedulerParameters=nohold_on_prolog_fail
-<% end -%>
```
If the decision to remove this is independent of the Slurm upgrade, I would split the commit in two (we can keep the commits in the same PR if you wish).

hanwen-pcluste replied:
Done

hanwen-cluster committed:
Upgrade Slurm to 22.05.3

In addition to the upgrade:
Remove TaskAffinity=no in cgroups.conf. This parameter had been ignored from Slurm 21. In addition, Slurm 22 fails if the parameter still exists.

Signed-off-by: Hanwen <hanwenli@amazon.com>

hanwen-cluster committed:
Remove prolog/epilog used as a workaround for cluster without Internet connection.

The Slurm bug has been fixed with Slurm 22.05. The workaround is no longer necessary

Signed-off-by: Hanwen <hanwenli@amazon.com>

demartinofra reviewed (approved):
I'm approving this but let's make sure we run some additional manual checks on scheduler and scaling logs

hanwen-pcluste committed:
Merge branch 'develop' into develop
